x-pack/plugins/observability_solution/apm/server/routes/assets/services/get_services_from_assets.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { termQuery, rangeQuery, kqlQuery } from '@kbn/observability-plugin/server';
+import { ASSET_TYPE, FIRST_SEEN, LAST_SEEN } from '../../../../common/es_fields/assets';
+import { AssetsESClient } from '../../../lib/helpers/create_es_client/create_assets_es_client/create_assets_es_clients';
+import { withApmSpan } from '../../../utils/with_apm_span';
+
+export const MAX_NUMBER_OF_SERVICES = 1_000;
+
+export async function getServicesFromAssets({
+  assetsESClient,
+  start,
+  end,
+  kuery,
+}: {
+  assetsESClient: AssetsESClient;
+  start: number;
+  end: number;
+  kuery: string;
+}) {
+  return withApmSpan('get_services_from_assets', async () => {
+    const response = await assetsESClient.search('get_services_from_assets', {
+      body: {
+        size: MAX_NUMBER_OF_SERVICES,
+        track_total_hits: false,
+        query: {
+          bool: {
+            filter: [...termQuery(ASSET_TYPE, 'service'), ...kqlQuery(kuery)],
+            should: [...rangeQuery(start, end, FIRST_SEEN), ...rangeQuery(start, end, LAST_SEEN)],
+          },
+        },
+      },
+    });
+
+    const services = response.hits.hits.map((hit) => ({
+      ...hit._source?.service,
+      signalTypes: hit._source?.asset?.signalTypes,
+      identifyingMetadata: hit._source?.asset?.identifying_metadata,
+    }));
```

Comment:
Maybe we should not merge assets fields with services fields?

Reply:
You're right.  I don't have a good reason why I decided to merge it 😄 

```
  "services": [{
    "asset": {
      "id": "synth-node-0",
      "identifying_metadata": [
        "service.name"
      ],
      "signalTypes": {
        "asset.traces": true,
        "asset.logs": true
      }
    },
    "service": {
      "environment": "Synthtrace: traces_logs_assets",
      "name": "synth-node-0",
    }
  }]
}
```

What about this ^

Reply:
We can come back to it if we feel this is not the right approach. Thanks for changing it.

Reply:
updated here: https://github.com/elastic/kibana/pull/181842/commits/ee9ec0c80203039a9919961f8174cc783165444e